pkg/pillar/cmd/nim/nim.go

```diff
@@ -782,6 +805,12 @@ func (n *nim) handleDPCImpl(key string, configArg interface{}, fromFile bool) {
 		n.forceLastResort = false
 		n.reevaluateLastResortDPC()
 	}
+	// if device can connect to controller it may get a new DPC in global config. This global DPC
+	// will have higher priority but can be invalid and the device will loose connectivity again
+	// at least temporarily while DPC is being tested. To avoid this we reset the timestamp on
+	// the Manual DPC to the current time
```

Comment:
If you do this when will you test the DPC received from the controller?
The answer can't be "never".

Reply:
Ok